lib/defaultingress/broker_ingress.go

```diff
@@ -60,8 +60,20 @@ func EnsureBrokerIngress(cr *ingressv1alpha1.AstarteDefaultIngress, parent *apiv
 		if e := controllerutil.SetControllerReference(cr, brokerService, scheme); e != nil {
 			return e
 		}
+		brokerService.Spec.Selector = map[string]string{"app": fmt.Sprintf("%s-vernemq", cr.Spec.Astarte)}
+		brokerService.Spec.Ports = []v1.ServicePort{
+			{
+				Port:       int32(pointy.Int16Value(parent.Spec.VerneMQ.Port, 8883)),
+				TargetPort: intstr.FromInt(8883),
+			},
+		}
+		brokerService.Spec.Type = cr.Spec.Broker.ServiceType
+		// required to preserve client IP
+		brokerService.Spec.ExternalTrafficPolicy = v1.ServiceExternalTrafficPolicyTypeLocal
 
-		brokerService.Spec = getBrokerServiceSpec(cr, parent)
+		if cr.Spec.Broker.LoadBalancerIP != "" && cr.Spec.Broker.ServiceType == v1.ServiceTypeLoadBalancer {
+			brokerService.Spec.LoadBalancerIP = cr.Spec.Broker.LoadBalancerIP
+		}
```

Comment:
Is there a reason why the `getBrokerServiceSpec` function was removed to move its implementation here?

Reply:
Got it